File: clients/python/examples/collections-find-one-and-replace-vectorize.py
from astrapy import DataAPIClient

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Replace a document
result = collection.find_one_and_replace(
    {},
    {"name": "Jane Doe", "age": 42},
    sort={"$vectorize": "Text to vectorize"},
)

print(result)

File: clients/java/examples/collections-find-one-and-replace-vectorize.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.options.CollectionFindOneAndReplaceOptions;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Sort;
import java.util.Optional;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Replace a document
    Document newDocument = new Document().append("name", "Jane Doe").append("age", 42);
    CollectionFindOneAndReplaceOptions options =
        new CollectionFindOneAndReplaceOptions().sort(Sort.vectorize("Text to vectorize"));
    Optional<Document> result = collection.findOneAndReplace(null, newDocument, options);
    System.out.println(result);
  }
}
